resources/views/includes/follow-button.blade.php

```diff
@@ -1,21 +1,25 @@
 @if($user->id == Auth::user()->id)
     <a href="{{ route('settings') }}" class="btn btn-sm btn-primary">{{ __('profile.settings') }}</a>
-@elseif($user->private_profile && $user->followRequests->contains('user_id', Auth::user()->id))
+@elseif(($user->private_profile || $user->follower_approval) && $user->followRequests->contains('user_id', Auth::user()->id))
```

Comment:
might be better if we introduced an attribute that generally evaluates whether follows of a user should be approved.